tests/test_console.py

```diff
@@ -368,6 +368,19 @@ def test_capture_and_record(capsys):
     assert out == "ABC\n"
 
 
+def test_capture_echo_outputs_captured_content_to_terminal(capsys):
```

Comment:
Could you add a test with more nesting. e..g. a capture inside a capture. Or a simple `with console:` block within a capture.

Reply:
Tried this test passes on `master` (it shouldn't), so I'm not sure nested captures work at all:

```python
def test_capture_echo_nested_capture():
    console = Console()
    console.print(1)
    with console.capture() as capture1:
        console.print(2)
        with console.capture() as capture2:
            console.print(3)
    console.print(4)

    assert capture1.get() == ""
    assert capture2.get() == ""
```

Reply:
I'm wondering what the expected behaviour should be.

The outer capture prevents content from getting to the terminal. So the inner capture has nothing to capture? So maybe this is the expected behaviour.

Could you also test this:

```python
with console.capture():
    with console:
        console.print("foo")
```

The with console should make the block atomic, i.e. it saves up all content until it exists the block.

Reply:
> The outer capture prevents content from getting to the terminal. So the inner capture has nothing to capture? So maybe this is the expected behaviour.

I'd still expect the `2` to be captured inside `capture1`.

This test passes:

```
def test_capture_echo_nested_capture_console():
    console = Console()
    with console.capture() as capture1:
        console.print(1)
        with console:
            console.print(2)

    assert capture1.get() == "1\n2\n"
```

Reply:
Ok, now that I've refreshed my memory with this stuff 😄 ...

This is the behaviour on `master`, which looks wrong:

```python
def test_capture_echo_nested_capture():
    console = Console()
    console.print(1)
    with console.capture() as capture1:
        console.print(2)
        with console.capture() as capture2:
            console.print(3)
    console.print(4)

    assert capture1.get() == ""
    assert capture2.get() == "2\n3\n"
```

My expectation would be that `capture1` captures `2\n3\n` and `capture2` captures `3\n`.